tests/TestCase/Utility/HashTest.php

```diff
@@ -1476,6 +1476,17 @@ public function testInsertMulti()
             4 => ['Item' => ['id' => 5, 'title' => 'fifth']],
         ];
         $this->assertEquals($expected, $result);
+
+        $data[3]['testable'] = true;
+        $result = Hash::insert($data, '{n}[testable].Item[id=/\b2|\b4/].test', 2);
```

Comment:
My own qualm with this is that this 'feature' is not implemented in other methods that use the hash path syntax. This introduces inconsistencies that make using Hash harder as the syntax harder to use as people have to remember which selectors are supported where.